src/runtime-rs/crates/hypervisor/src/qemu/network.rs

```diff
@@ -0,0 +1,372 @@
+            | NetDeviceType::MacvTap
+            | NetDeviceType::IpvTap => NetDeviceDriver::VirtioNet,
+            NetDeviceType::VhuNet => {
+                if transport == &VirtioTransport::Ccw {
+                    return Err(anyhow!("vhost-user devices are not supported on IBM Z"));
+                } else {
+                    NetDeviceDriver::Vhostuser
+                }
+            }
+            NetDeviceType::Physical => {
+                if transport == &VirtioTransport::Mmio {
+                    return Err(anyhow!(
+                        "vfio devices are not support with the MMIO transport"
+                    ));
+                } else {
+                    NetDeviceDriver::Vfio
+                }
+            }
+        };
+
+        Ok(format!("{}-{}", driver.to_string(), transport.to_string()))
+    }
+}
+
+impl ToString for NetDeviceType {
+    fn to_string(&self) -> String {
+        match self {
+            NetDeviceType::Tap
+            | NetDeviceType::MacvTap
+            | NetDeviceType::IpvTap
+            | NetDeviceType::VethTap => "tap".to_owned(),
+            NetDeviceType::VhuNet => "vhostuser".to_owned(),
+            NetDeviceType::Physical => "vfio".to_owned(),
+        }
+    }
+}
+
+// VirtioTransport is the transport in use for a virtio device.
+#[derive(Debug, Default, PartialEq)]
+enum VirtioTransport {
+    #[default]
+    Pci,
+    Ccw,
+    Mmio,
+    Ap,
+}
+
+impl ToString for VirtioTransport {
+    fn to_string(&self) -> String {
+        match self {
+            VirtioTransport::Pci => "pci".to_owned(),
+            VirtioTransport::Ccw => "ccw".to_owned(),
+            VirtioTransport::Mmio => "mmio".to_owned(),
+            VirtioTransport::Ap => "ap".to_owned(),
+        }
+    }
+}
+
+// -netdev tap,fd=37,id=hostnet0,vhost=on,vhostfd=38
+// -device virtio-net-pci,netdev=hostnet0,id=net0,mac=24:42:54:20:50:46,bus=pci.0,addr=0x7
+// NetDevice represents a guest networking device
+#[derive(Debug, Default)]
+pub struct NetDevice {
+    // Type is the netdev type (e.g. tap).
+    net_type: NetDeviceType,
+
+    // Driver is the qemu device driver
+    driver: NetDeviceDriver,
+
+    // ID is the netdevice identifier.
+    id: String,
+
+    // IfName is the interface name,
+    if_name: String,
+
+    // Bus is the bus path name of a PCI device.
+    bus: String,
+
+    // Addr is the address offset of a PCI device.
+    pci_addr: String,
+
+    // DownScript is the tap interface deconfiguration script.
+    down_script: Option<String>,
+
+    // Script is the tap interface configuration script.
+    script: Option<String>,
+
+    // FIXME: to support mq
+    // // FDs represents the list of already existing file descriptors to be used.
+    // // This is mostly useful for mq support.
+    // fds:      Vec<File>,
+    // vhost_fds: Vec<File>,
+
+    // VHost enables virtio device emulation from the host kernel instead of from qemu.
+    vhost_enabled: bool,
```

Comment:
`configuration.toml` doesn't allow users to set this per device so I think we should simplify by dropping this and referring to the master copy of this piece of information in `HypervisorConfig::network_info::disable_vhost_net` instead.